### ISSUE TITLE
APMRover: add a way to dectect a broken steering

### ISSUE THREAD

**Issue by khancyr** [ArduPilot/ardupilot#6107]
Idea from http://discuss.ardupilot.org/t/what-events-trigger-fs-crash-check/16697
Could potentialy be added to the crash check

**Comment by rmackay9** [comment]
This idea came up again in this discussion during Rover-4.1.0-beta testing https://discuss.ardupilot.org/t/yuris-4-1-0-beta-update-gps-yaw-now-beta3/71100/118.

**Comment by rmackay9** [comment]
This issue came up again in a discussion with a Partner on 21-Dec-2021

**Comment by geofrancis** [comment]
@rmackay9  Would monitoring the difference between piddesired and pidachieved be enough to catch a steering failure?

**Comment by rmackay9** [comment]
@geofrancis,

Basically yes I think that should work.  We actually have (in copter at least) a mavlink message that we publish which has the root-mean-squared difference between the two.  I don't think anything actually makes use of these calculations but this is probably the basis of what we should use for the control failsafe.

**Comment by geofrancis** [comment]
@rmackay9 a boat could get into a situation where its some kind of environmental issue like the wind stopping it turning that could be interpreted as a steering failure. you would probably want some kind of weighting to it or at least a decent timeout so it doesn't give false alerts. There is the code in sail that checks if the boat is too far off course due to a failed tack and engages the motors that could be useful.

**Comment by rmackay9** [comment]
@geofrancis,

Yes, similar to the FS_EKF_THRESH which allows setting the sensitivity of the failsafe..

**Comment by geofrancis** [comment]
@rmackay9  yes, for something like a boat thats not going to have much around it could probably be relaxed a lot, compared to something like a lawnmower where you want much tighter tolerances.
